### PR TITLE
FvBaseDiscretization: add serialization support

### PR DESCRIPTION
**Pull request by akva2**
Sits on top of https://github.com/OPM/opm-models/pull/779

For restart.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-common=3383 opm-simulators=4422 please

**Comment by bska** [comment]
> Sits on top of #779

That PR was merged.  Is this PR still a work in progress?

**Comment by akva2** [comment]
No, just wasn't pertient yet since the need is a bit down the chain in simulators still.

**Comment by bska** [comment]
jenkins build this please

File: opm/models/discretization/common/fvbasediscretization.hh [hunk 1]
```diff
@@ -388,12 +389,44 @@ class FvBaseDiscretization
...
+
+            return result;
+        }
+
         SolutionVector& blockVector()
         { return blockVector_; }
         const SolutionVector& blockVector() const
         { return blockVector_; }
+
+        bool operator==(const BlockVectorWrapper& wrapper)
```
Comment (bska) [inline opm/models/discretization/common/fvbasediscretization.hh:409]: The equality operator should typically be marked `const`.

File: opm/models/discretization/common/fvbasediscretization.hh [hunk 2]
```diff
@@ -1869,6 +1902,27 @@ public:
...
+        if (this->solution_.size() != rhs.solution_.size())
+            return false;
+
+        for (size_t i = 0; i < this->solution_.size(); ++i) {
+            if (*this->solution_[i] != *rhs.solution_[i]) {
+                return false;
+            }
+        }
+
+        return true;
```
Comment (bska) [inline opm/models/discretization/common/fvbasediscretization.hh:1923]: I **suppose** you could write this in terms of [`std::equal()`](https://en.cppreference.com/w/cpp/algorithm/equal)
```C++
return std::equal(this->solution_.begin(), this->solution_.end(),
                  rhs.solution_.begin(), rhs.solution_.end(),
    [](const auto& x, const auto& y)
    {
        return *x == *y;
    });
```
Whether or not that's more readable is probably in the eyes of the beholder.

**Review by bska — COMMENTED** [review]
Looks good for the most part.  One small nit and one suggestion for you to consider.  Otherwise this is ready for merging.

File: opm/models/discretization/common/fvbasediscretization.hh (hunk 2 above)
Comment (akva2) [reply opm/models/discretization/common/fvbasediscretization.hh]: i think it is more readable, one quick glance and you can see what's going on. changed both operators to use it.

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  I appreciate using `std::equal()` in both equality operators.  I too think that's more readable than a raw loop.  I'll merge into master once the build check succeeds.

**Comment by bska** [comment]
> I'll merge into master once the build check succeeds.

Build check is successful.  I'll merge into master.

**Pull request merged**